File: leetcode/easy/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (map.containsKey(diff)) {
                int j = map.get(diff);
                return new int[] { j, i };
            } else {
                map.put(nums[i], i);
            }
        }
        return null;
    }
}

File: leetcode/easy/TwoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mapping = {}
        for i, num in enumerate(nums):
            if target - num in mapping:
                return [mapping.get(target - num), i]
            else:
                mapping[num] = i
